Author: reimar
Date: Sat Jan 18 22:39:07 2014
New Revision: 36613

Log:
Remove empty draw_slice and draw_frame stubs.

Long term, draw_frame should be removed completely.

NOTE: patch modified to include change from r36723, but exclude changes
      to libvo/vosub_vidix.c

Modified:
   trunk/libvo/mga_template.c
   trunk/libvo/vo_caca.c
   trunk/libvo/vo_cvidix.c
   trunk/libvo/vo_dfbmga.c
   trunk/libvo/vo_dga.c
   trunk/libvo/vo_direct3d.c
   trunk/libvo/vo_directfb2.c
   trunk/libvo/vo_dxr2.c
   trunk/libvo/vo_dxr3.c
   trunk/libvo/vo_fbdev.c
   trunk/libvo/vo_fbdev2.c
   trunk/libvo/vo_gif89a.c
   trunk/libvo/vo_gl.c
   trunk/libvo/vo_ivtv.c
   trunk/libvo/vo_jpeg.c
   trunk/libvo/vo_kva.c
   trunk/libvo/vo_matrixview.c
   trunk/libvo/vo_md5sum.c
   trunk/libvo/vo_mga.c
   trunk/libvo/vo_mng.c
   trunk/libvo/vo_mpegpes.c
   trunk/libvo/vo_null.c
   trunk/libvo/vo_png.c
   trunk/libvo/vo_pnm.c
   trunk/libvo/vo_s3fb.c
   trunk/libvo/vo_svga.c
   trunk/libvo/vo_tga.c
   trunk/libvo/vo_v4l2.c
   trunk/libvo/vo_vdpau.c
   trunk/libvo/vo_wii.c
   trunk/libvo/vo_winvidix.c
   trunk/libvo/vo_x11.c
   trunk/libvo/vo_xmga.c
   trunk/libvo/vo_xv.c
   trunk/libvo/vo_xvidix.c
   trunk/libvo/vo_xvmc.c
   trunk/libvo/vo_xvr100.c
   trunk/libvo/vo_yuv4mpeg.c
   trunk/libvo/vo_zr2.c
   trunk/libvo/vosub_vidix.c

Modified: trunk/libvo/mga_template.c
==============================================================================
--- trunk/libvo/mga_template.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/mga_template.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -140,13 +140,6 @@ vo_mga_flip_page(void)
 
 }
 
-static int
-draw_frame(uint8_t *src[])
-{
-    mp_msg(MSGT_VO,MSGL_WARN,"!!! mga::draw_frame() called !!!\n");
-    return 0;
-}
-
 static uint32_t get_image(mp_image_t *mpi){
     uint32_t bespitch = FFALIGN(mga_vid_config.src_width, 32);
     uint32_t bespitch2 = bespitch/2;

Modified: trunk/libvo/vo_caca.c
==============================================================================
--- trunk/libvo/vo_caca.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_caca.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -35,6 +35,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "sub/sub.h"
 
@@ -178,11 +179,6 @@ static int draw_frame(uint8_t *src[])
     return 0;
 }
 
-static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
-{
-    return 0;
-}
-
 static void flip_page(void)
 {
     if (showosdmessage) {

Modified: trunk/libvo/vo_cvidix.c
==============================================================================
--- trunk/libvo/vo_cvidix.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_cvidix.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -30,6 +30,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "aspect.h"
 #include "geometry.h"
@@ -121,7 +122,7 @@ static int config(uint32_t width, uint32
 static void check_events(void){
 }
 
-/* draw_osd, flip_page, draw_slice, draw_frame should be
+/* draw_osd, flip_page, draw_slice should be
    overwritten with vidix functions (vosub_vidix.c) */
 static void draw_osd(void){
   mp_msg(MSGT_VO, MSGL_FATAL, "vo_cvidix: error: didn't use vidix draw_osd!\n");
@@ -138,11 +139,6 @@ static int draw_slice(uint8_t *src[], in
   return -1;
 }
 
-static int draw_frame(uint8_t *src[]){
-  mp_msg(MSGT_VO, MSGL_FATAL, "vo_cvidix: error: didn't use vidix draw_frame!\n");
-  return -1;
-}
-
 static int query_format(uint32_t format){
   return vidix_query_fourcc(format);
 }

Modified: trunk/libvo/vo_dfbmga.c
==============================================================================
--- trunk/libvo/vo_dfbmga.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_dfbmga.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -30,6 +30,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "fastmemcpy.h"
@@ -1047,12 +1048,6 @@ draw_alpha( int x0, int y0,
 }
 
 static int
-draw_frame( uint8_t * src[] )
-{
-     return -1;
-}
-
-static int
 draw_slice( uint8_t * src[], int stride[], int w, int h, int x, int y )
 {
      uint8_t *dst;

Modified: trunk/libvo/vo_dga.c
==============================================================================
--- trunk/libvo/vo_dga.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_dga.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -28,6 +28,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "aspect.h"
 #include "x11_common.h"
@@ -346,14 +347,6 @@ static void flip_page(void)
 
 //---------------------------------------------------------
 
-static int draw_slice(uint8_t * src[], int stride[],
-                           int w, int h, int x, int y)
-{
-    return 0;
-};
-
-//---------------------------------------------------------
-
 static int query_format(uint32_t format)
 {
 

Modified: trunk/libvo/vo_direct3d.c
==============================================================================
--- trunk/libvo/vo_direct3d.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_direct3d.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -24,6 +24,7 @@
 #include <d3d9.h>
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "fastmemcpy.h"
 #include "mp_msg.h"
@@ -972,14 +973,6 @@ static int draw_slice(uint8_t *src[], in
     return 0; /* Success */
 }
 
-/** @brief libvo Callback: Unused function
- */
-static int draw_frame(uint8_t *src[])
-{
-    mp_msg(MSGT_VO, MSGL_V, "<vo_direct3d>draw_frame called.\n");
-    return VO_FALSE;
-}
-
 /** @brief Maps MPlayer alpha to D3D
  *         0x0 -> transparent and discarded by alpha test
  *         0x1 -> 0xFF to become opaque

Modified: trunk/libvo/vo_directfb2.c
==============================================================================
--- trunk/libvo/vo_directfb2.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_directfb2.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -31,6 +31,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "fastmemcpy.h"
@@ -1395,15 +1396,6 @@ static int control(uint32_t request, voi
   return VO_NOTIMPL;
 }
 
-// unused function
-
-static int draw_frame(uint8_t *src[])
-{
-	return -1;
-}
-
-// hopefully will be removed soon
-
 static void draw_alpha(int x0, int y0, int w, int h, unsigned char *src,
 		unsigned char *srca, int stride)
 {

Modified: trunk/libvo/vo_dxr2.c
==============================================================================
--- trunk/libvo/vo_dxr2.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_dxr2.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -33,6 +33,7 @@
 #include "config.h"
 #include "aspect.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "mp_msg.h"
 #include "m_option.h"
@@ -761,11 +762,6 @@ static void flip_page (void)
     sub_vo->flip_page();
 }
 
-static int draw_slice( uint8_t *srcimg[], int stride[], int w, int h, int x0, int y0 )
-{
-  return 0;
-}
-
 
 static int query_format(uint32_t format)
 {

Modified: trunk/libvo/vo_dxr3.c
==============================================================================
--- trunk/libvo/vo_dxr3.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_dxr3.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -41,6 +41,7 @@
 #include "fastmemcpy.h"
 
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "aspect.h"
@@ -675,11 +676,6 @@ static void flip_page(void)
 	}
 }
 
-static int draw_slice(uint8_t *srcimg[], int stride[], int w, int h, int x0, int y0)
-{
-	return -1;
-}
-
 static void uninit(void)
 {
 	mp_msg(MSGT_VO,MSGL_INFO, MSGTR_LIBVO_DXR3_Uninitializing);

Modified: trunk/libvo/vo_fbdev.c
==============================================================================
--- trunk/libvo/vo_fbdev.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_fbdev.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -41,6 +41,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "fastmemcpy.h"
 #include "sub/sub.h"
@@ -1054,11 +1055,6 @@ static void draw_alpha(int x0, int y0, i
     (*draw_alpha_p)(w, h, src, srca, stride, dst, fb_line_len);
 }
 
-static int draw_frame(uint8_t *src[])
-{
-    return 1;
-}
-
 static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
 {
     uint8_t *d;

Modified: trunk/libvo/vo_fbdev2.c
==============================================================================
--- trunk/libvo/vo_fbdev2.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_fbdev2.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -32,6 +32,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "fastmemcpy.h"
 #include "sub/sub.h"
@@ -371,9 +372,6 @@ static void draw_osd(void)
 	vo_draw_text(in_width, in_height, draw_alpha);
 }
 
-// all csp support stride
-static int draw_frame(uint8_t *src[]) { return 1; }
-
 static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
 {
 	uint8_t *in = src[0];

Modified: trunk/libvo/vo_gif89a.c
==============================================================================
--- trunk/libvo/vo_gif89a.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_gif89a.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -54,6 +54,7 @@
 #include "config.h"
 #include "subopt-helper.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "mp_msg.h"
 
@@ -322,11 +323,6 @@ static void flip_page(void)
 	EGifPutLine(new_gif, reduce_data, img_width * img_height);
 }
 
-static int draw_frame(uint8_t *src[])
-{
-	return 1;
-}
-
 static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
 {
 	uint8_t *dst, *frm;

Modified: trunk/libvo/vo_gl.c
==============================================================================
--- trunk/libvo/vo_gl.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_gl.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -31,6 +31,7 @@
 #include "mp_msg.h"
 #include "subopt-helper.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "sub/font_load.h"
@@ -1135,12 +1136,6 @@ skip_upload:
 }
 
 static int
-draw_frame(uint8_t *src[])
-{
-  return VO_ERROR;
-}
-
-static int
 query_format(uint32_t format)
 {
     int depth;

Modified: trunk/libvo/vo_ivtv.c
==============================================================================
--- trunk/libvo/vo_ivtv.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_ivtv.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -45,6 +45,7 @@
 #include "mp_msg.h"
 #include "subopt-helper.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "vo_ivtv.h"
 #include "libmpdemux/mpeg_packetizer.h"
@@ -250,12 +251,6 @@ flip_page (void)
   pes = NULL;
 }
 
-static int
-draw_slice (uint8_t *image[], int stride[], int w, int h, int x, int y)
-{
-  return 0;
-}
-
 static void
 uninit (void)
 {

Modified: trunk/libvo/vo_jpeg.c
==============================================================================
--- trunk/libvo/vo_jpeg.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_jpeg.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -43,6 +43,7 @@
 #include "subopt-helper.h"
 #include "mp_msg.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "mp_core.h"
 #include "help_mp.h"
@@ -287,14 +288,6 @@ static void flip_page (void)
 
 /* ------------------------------------------------------------------------- */
 
-static int draw_slice(uint8_t *src[], int stride[], int w, int h,
-                           int x, int y)
-{
-    return 0;
-}
-
-/* ------------------------------------------------------------------------- */
-
 static int query_format(uint32_t format)
 {
     if (format == IMGFMT_RGB24) {

Modified: trunk/libvo/vo_kva.c
==============================================================================
--- trunk/libvo/vo_kva.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_kva.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -39,6 +39,7 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "aspect.h"
@@ -1004,11 +1005,6 @@ static int control(uint32_t request, voi
     return VO_NOTIMPL;
 }
 
-static int draw_frame(uint8_t *src[])
-{
-    return VO_ERROR;
-}
-
 static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
 {
     uint8_t *s;

Modified: trunk/libvo/vo_matrixview.c
==============================================================================
--- trunk/libvo/vo_matrixview.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_matrixview.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -30,6 +30,7 @@
 #include "mp_msg.h"
 #include "subopt-helper.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "gl_common.h"
@@ -173,12 +174,6 @@ static int draw_slice(uint8_t *src[], in
 }
 
 
-static int draw_frame(uint8_t *src[])
-{
-    return 0;
-}
-
-
 static int query_format(uint32_t format)
 {
     int caps = VFCAP_CSP_SUPPORTED | VFCAP_HWSCALE_UP | VFCAP_HWSCALE_DOWN | VFCAP_ACCEPT_STRIDE;

Modified: trunk/libvo/vo_md5sum.c
==============================================================================
--- trunk/libvo/vo_md5sum.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_md5sum.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -39,6 +39,8 @@
 #include "subopt-helper.h"
 #include "mp_msg.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "mp_core.h"			/* for exit_player() */
 #include "help_mp.h"
@@ -180,14 +182,6 @@ static void md5sum_output_sum(unsigned c
 
 /* ------------------------------------------------------------------------- */
 
-static int draw_frame(uint8_t *src[])
-{
-    mp_msg(MSGT_VO, MSGL_V, "%s: draw_frame() is called!\n", info.short_name);
-    return -1;
-}
-
-/* ------------------------------------------------------------------------- */
-
 static uint32_t draw_image(mp_image_t *mpi)
 {
     unsigned char md5sum[16];
@@ -241,14 +235,6 @@ static uint32_t draw_image(mp_image_t *m
 
 /* ------------------------------------------------------------------------- */
 
-static int draw_slice(uint8_t *src[], int stride[], int w, int h,
-                           int x, int y)
-{
-    return 0;
-}
-
-/* ------------------------------------------------------------------------- */
-
 static int query_format(uint32_t format)
 {
     switch (format) {

Modified: trunk/libvo/vo_mga.c
==============================================================================
--- trunk/libvo/vo_mga.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_mga.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -26,6 +26,7 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 
 #include <sys/ioctl.h>

Modified: trunk/libvo/vo_mng.c
==============================================================================
--- trunk/libvo/vo_mng.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_mng.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -38,6 +38,7 @@
 #include <libmng.h>
 
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "mp_msg.h"
 #include "subopt-helper.h"
@@ -514,18 +515,6 @@ static void flip_page(void)
 }
 
 /**
- * @brief put frame data onto canvas (not supported)
- * @return always 1 to indicate error
- */
-static int draw_frame(uint8_t *src[])
-{
-    /* draw_frame() not supported
-     * VFCAP_ACCEPT_STRIDE is set for format
-     * so draw_slice() will be called instead of this function */
-    return 1;
-}
-
-/**
  * @brief deinitialize MNG vo module
  */
 static void uninit(void)

Modified: trunk/libvo/vo_mpegpes.c
==============================================================================
--- trunk/libvo/vo_mpegpes.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_mpegpes.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -54,6 +54,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "libmpdemux/mpeg_packetizer.h"
 
@@ -215,12 +216,6 @@ static void flip_page (void)
 {
 }
 
-static int draw_slice(uint8_t *srcimg[], int stride[], int w,int h,int x0,int y0)
-{
-    return 0;
-}
-
-
 static int
 query_format(uint32_t format)
 {

Modified: trunk/libvo/vo_null.c
==============================================================================
--- trunk/libvo/vo_null.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_null.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -28,6 +28,7 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 
 static const vo_info_t info =
@@ -59,12 +60,6 @@ flip_page(void)
 }
 
 static int
-draw_frame(uint8_t *src[])
-{
-	return 0;
-}
-
-static int
 query_format(uint32_t format)
 {
     if (IMGFMT_IS_HWACCEL(format))

Modified: trunk/libvo/vo_png.c
==============================================================================
--- trunk/libvo/vo_png.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_png.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -36,6 +36,8 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "subopt-helper.h"
 #include "libavcodec/avcodec.h"
@@ -191,16 +193,6 @@ static void draw_osd(void){}
 
 static void flip_page (void){}
 
-static int draw_frame(uint8_t * src[])
-{
-    return -1;
-}
-
-static int draw_slice( uint8_t *src[],int stride[],int w,int h,int x,int y )
-{
-    return -1;
-}
-
 static int
 query_format(uint32_t format)
 {

Modified: trunk/libvo/vo_pnm.c
==============================================================================
--- trunk/libvo/vo_pnm.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_pnm.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -38,6 +38,8 @@
 #include "subopt-helper.h"
 #include "mp_msg.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "mp_core.h"			/* for exit_player() */
 #include "help_mp.h"
@@ -504,22 +506,6 @@ static uint32_t draw_image(mp_image_t *m
 
 /* ------------------------------------------------------------------------- */
 
-static int draw_frame(uint8_t *src[])
-{
-    mp_msg(MSGT_VO, MSGL_V, "%s: draw_frame() is called!\n", info.short_name);
-    return -1;
-}
-
-/* ------------------------------------------------------------------------- */
-
-static int draw_slice(uint8_t *src[], int stride[], int w, int h,
-                           int x, int y)
-{
-    return 0;
-}
-
-/* ------------------------------------------------------------------------- */
-
 static int query_format(uint32_t format)
 {
     /* Ensure that for PPM we get Packed RGB and for PGM(YUV) we get

Modified: trunk/libvo/vo_s3fb.c
==============================================================================
--- trunk/libvo/vo_s3fb.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_s3fb.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -42,6 +42,7 @@
 #include "mp_msg.h"
 #include "fastmemcpy.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "aspect.h"
 #include "sub/sub.h"
@@ -493,11 +494,6 @@ static int draw_frame(uint8_t *src[])
   return 0;
 }
 
-static int draw_slice(uint8_t *i[], int s[], int w, int h, int x, int y)
-{
-  return 1;
-}
-
 /* Attempt to start doing DR */
 static uint32_t get_image(mp_image_t *mpi)
 {

Modified: trunk/libvo/vo_svga.c
==============================================================================
--- trunk/libvo/vo_svga.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_svga.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -52,6 +52,8 @@ TODO:
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "fastmemcpy.h"
 #include "osdep/getch2.h"
@@ -553,17 +555,6 @@ static int config(uint32_t width, uint32
     return 0;
 }
 
-static int draw_slice(uint8_t *image[],int stride[],
-               int w, int h, int x, int y) {
-    assert(0);
-    return VO_ERROR;//this is yv12 only -> vf_scale should do all transforms
-}
-
-static int draw_frame(uint8_t *src[]) {
-    assert(0);
-    return VO_ERROR;//this one should not be called
-}
-
 static void draw_osd(void)
 {
     mp_msg(MSGT_VO,MSGL_DBG4, "vo_svga: draw_osd()\n");

Modified: trunk/libvo/vo_tga.c
==============================================================================
--- trunk/libvo/vo_tga.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_tga.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -53,6 +53,8 @@
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 
 static const vo_info_t info =
@@ -181,16 +183,6 @@ static void flip_page (void)
     return;
 }
 
-static int draw_slice(uint8_t *srcimg[], int stride[], int w,int h,int x,int y)
-{
-    return -1;
-}
-
-static int draw_frame(uint8_t * src[])
-{
-    return -1;
-}
-
 static int query_format(uint32_t format)
 {
     switch(format){

Modified: trunk/libvo/vo_v4l2.c
==============================================================================
--- trunk/libvo/vo_v4l2.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_v4l2.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -40,6 +40,7 @@
 #include "mp_msg.h"
 #include "subopt-helper.h"
 #include "video_out.h"
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "libmpdemux/mpeg_packetizer.h"
 #include "vo_v4l2.h"
@@ -221,12 +222,6 @@ flip_page (void)
   pes = NULL;
 }
 
-static int
-draw_slice (uint8_t *image[], int stride[], int w, int h, int x, int y)
-{
-  return 0;
-}
-
 static void
 uninit (void)
 {

Modified: trunk/libvo/vo_vdpau.c
==============================================================================
--- trunk/libvo/vo_vdpau.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_vdpau.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -39,6 +39,7 @@
 #include "sub/ass_mp.h"
 #include "mp_msg.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "x11_common.h"
@@ -1042,11 +1043,6 @@ static int draw_slice(uint8_t *image[], 
 }
 
 
-static int draw_frame(uint8_t *src[])
-{
-    return VO_ERROR;
-}
-
 static struct vdpau_render_state *get_surface(int number)
 {
     if (number >= MAX_VIDEO_SURFACES)

Modified: trunk/libvo/vo_wii.c
==============================================================================
--- trunk/libvo/vo_wii.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_wii.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -50,6 +50,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "sub/sub.h"
 #include "mp_msg.h"
@@ -268,11 +269,6 @@ static void draw_alpha(int x0, int y0, i
   vo_draw_alpha_yuy2(w, h, src, srca, stride, dst, fb_line_len);
 }
 
-static int draw_frame(uint8_t *src[])
-{
-  return 1;
-}
-
 static int draw_slice(uint8_t *src[], int stride[], int w, int h, int x, int y)
 {
   uint8_t *d, *s;

Modified: trunk/libvo/vo_winvidix.c
==============================================================================
--- trunk/libvo/vo_winvidix.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_winvidix.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -28,6 +28,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 
 #include <windows.h>
@@ -279,7 +280,7 @@ static void check_events(void){
     }
 }
 
-/* draw_osd, flip_page, draw_slice, draw_frame should be
+/* draw_osd, flip_page, draw_slice should be
    overwritten with vidix functions (vosub_vidix.c) */
 static void draw_osd(void){
     mp_msg(MSGT_VO, MSGL_FATAL, "[winvidix] error: didn't use vidix draw_osd!\n");
@@ -296,11 +297,6 @@ static int draw_slice(uint8_t *src[], in
     return -1;
 }
 
-static int draw_frame(uint8_t *src[]){
-    mp_msg(MSGT_VO, MSGL_FATAL, "[winvidix] error: didn't use vidix draw_frame!\n");
-    return -1;
-}
-
 static int query_format(uint32_t format){
   return vidix_query_fourcc(format);
 }

Modified: trunk/libvo/vo_x11.c
==============================================================================
--- trunk/libvo/vo_x11.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_x11.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -22,6 +22,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 #include "aspect.h"
@@ -495,11 +496,6 @@ static int draw_slice(uint8_t * src[], i
     return 0;
 }
 
-static int draw_frame(uint8_t * src[])
-{
-    return VO_ERROR;
-}
-
 static uint32_t get_image(mp_image_t * mpi)
 {
     if (zoomFlag ||

Modified: trunk/libvo/vo_xmga.c
==============================================================================
--- trunk/libvo/vo_xmga.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_xmga.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -27,6 +27,7 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 
 

Modified: trunk/libvo/vo_xv.c
==============================================================================
--- trunk/libvo/vo_xv.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_xv.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -42,6 +42,7 @@ Buffer allocation:
 #include "mp_msg.h"
 #include "help_mp.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "libmpcodecs/vf.h"
 
@@ -417,11 +418,6 @@ static int draw_slice(uint8_t * image[],
     return 0;
 }
 
-static int draw_frame(uint8_t * src[])
-{
-    return VO_ERROR;
-}
-
 static uint32_t draw_image(mp_image_t * mpi)
 {
     if (mpi->flags & MP_IMGFLAG_DIRECT)

Modified: trunk/libvo/vo_xvidix.c
==============================================================================
--- trunk/libvo/vo_xvidix.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_xvidix.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -32,6 +32,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 
 #include <X11/Xlib.h>
@@ -316,7 +317,7 @@ static void check_events(void)
     return;
 }
 
-/* draw_osd, flip_page, draw_slice, draw_frame should be
+/* draw_osd, flip_page, draw_slice should be
    overwritten with vidix functions (vosub_vidix.c) */
 static void draw_osd(void)
 {
@@ -340,13 +341,6 @@ static int draw_slice(uint8_t * src[], i
     return -1;
 }
 
-static int draw_frame(uint8_t * src[])
-{
-    mp_msg(MSGT_VO, MSGL_FATAL,
-           "[xvidix] error: didn't used vidix draw_frame!\n");
-    return -1;
-}
-
 static int query_format(uint32_t format)
 {
     return vidix_query_fourcc(format);

Modified: trunk/libvo/vo_xvmc.c
==============================================================================
--- trunk/libvo/vo_xvmc.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_xvmc.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -24,6 +24,7 @@
 #include "config.h"
 #include "mp_msg.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "osdep/timer.h"
 #include "libmpcodecs/vf.h"
@@ -729,10 +730,6 @@ skip_surface_allocation:
     return 0;
 }
 
-static int draw_frame(uint8_t *srcp[]){
-    assert(0);
-}
-
 static void init_osd_yuv_pal(void) {
     char * palette;
     int rez;

Modified: trunk/libvo/vo_xvr100.c
==============================================================================
--- trunk/libvo/vo_xvr100.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_xvr100.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -44,6 +44,7 @@
 
 #include "config.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 #include "aspect.h"
 #include "geometry.h"
@@ -369,12 +370,6 @@ static uint32_t get_image(mp_image_t *mp
     return VO_TRUE;
 }
 
-static int draw_frame(uint8_t *src[])
-{
-    mp_msg(MSGT_VO,MSGL_WARN,"!!! vo_xvr100::draw_frame() called !!!\n");
-    return 0;
-}
-
 
 static int draw_slice(uint8_t *src[], int stride[], int w,int h,int x,int y)
 {

Modified: trunk/libvo/vo_yuv4mpeg.c
==============================================================================
--- trunk/libvo/vo_yuv4mpeg.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_yuv4mpeg.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -47,6 +47,7 @@
 #include "config.h"
 #include "subopt-helper.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
 #include "video_out_internal.h"
 
 #include "mp_msg.h"
@@ -221,12 +222,6 @@ static int draw_slice(uint8_t *srcimg[],
 	return 0;
 }
 
-static int draw_frame(uint8_t * src[])
-{
-			// gets done in draw_slice
-    return 0;
-}
-
 static int query_format(uint32_t format)
 {
 	if (format == IMGFMT_YV12)

Modified: trunk/libvo/vo_zr2.c
==============================================================================
--- trunk/libvo/vo_zr2.c	Sat Jan 18 21:41:46 2014	(r36612)
+++ trunk/libvo/vo_zr2.c	Sat Jan 18 22:39:07 2014	(r36613)
@@ -38,6 +38,8 @@
 #include "config.h"
 #include "videodev_mjpeg.h"
 #include "video_out.h"
+#define NO_DRAW_FRAME
+#define NO_DRAW_SLICE
 #include "video_out_internal.h"
 #include "mp_msg.h"
 #include "subopt-helper.h"
@@ -461,15 +463,6 @@ static int control(uint32_t request, voi
 	return VO_NOTIMPL;
 }
 
-static int draw_frame(uint8_t *src[]) {
-	return 0;
-}
-
-static int draw_slice(uint8_t *image[], int stride[],
-		int w, int h, int x, int y) {
- 	return 0;
-}
-
 static void draw_osd(void) {
 }
 

